homeassistant/components/dsmr/sensor.py

```diff
@@ -137,44 +143,70 @@ def update_entities_telegram(telegram):
 
     async def connect_and_reconnect():
         """Connect to DSMR and keep reconnecting until Home Assistant stops."""
-        while hass.state != CoreState.stopping:
+        while True:
```

Comment:
this worries me a little, potential endless loop

Reply:
Hmm...I think it's not needed either, let me see if I can refactor it.

Reply:
It was actually returning inside the loop when `hass.state` was `CoreState.stopping`, so I put this back not to have an endless loop